lib/passports.js

```diff
@@ -105,14 +105,20 @@ module.exports = {
                 passport.tokens = JSON.stringify(query.tokens)
               }
 
-              // Save any updates to the Passport before moving on
-              app.services.FootprintService.update('Passport', passport.id, passport) .then(passport => {
-                app.services.FootprintService.find('user', passport[0].user.id)
-                  .then(user => {
-                    next(null, user)
-                  })
-                  .catch(next)
-              }).catch(next)
+              // I save any update to the Passport before moving on and
+              // I read the associated user instance
+              return Promise.all([
+                app.services.FootprintService.findAssociation("Passport", passport.id, "User"),
+                app.services.FootprintService.update("Passport", passport.id, passport)
+              ])
+                .then(results => {
+                  let userInstance = results[0]; // Not so usefull, just for keeping namings clear
+                  return app.services.FootprintService.find('user', userInstance.id)
```

Comment:
Why do you do a find again  ? You already have the user under `userInstance ` right ?